pkg/crd/gen.go

```diff
@@ -83,6 +83,9 @@ type Generator struct {
 	// You'll need to use "v1" to get support for features like defaulting,
 	// along with an API server that supports it (Kubernetes 1.16+).
 	CRDVersions []string `marker:"crdVersions,optional"`
+
+	// preserve object meta data
+	PreserveObjectMeta *bool `marker:",optional"`
```

Comment:
this flag seems to do the inverse of what i'd expect, if I'm reading this right

Here's what I'd expect:

if you pass PreserveObjectMeta, you get an object with no top-level metadata (it's enforced by the apiserver), but all the embedded object metadata fields are filled out.

If you don't pass PreserveObjectMeta, you get the behavior today, which is that *all* objectmeta fields are wiped.

Reply:
@DirectXMan12 thanks for taking a look ! I probably misunderstood but it had seemed when PreserveObjectMeta is either unspecified or set to false, with this update we would not invoke the new fnc FixupMetaData() and hence the behavior would be what we have today. I shall re-read and flip the check if that is what you meant.

Reply:
@DirectXMan12 I think the top leve metadata will be rejected by the API server regardless, I thought this PR was about keeping the metadata for embedded `ObjectMeta` in the CRDs

Reply:
cc @champak